unstructured/chunking/__init__.py

```diff
@@ -25,44 +26,70 @@ def add_chunking_strategy() -> Callable[[Callable[_P, List[Element]]], Callable[
                 + "\n\tAdditional Parameters:"
                 + "\n\t\tmultipage_sections"
                 + "\n\t\t\tIf True, sections can span multiple pages. Defaults to True."
                 + "\n\t\tcombine_text_under_n_chars"
                 + "\n\t\t\tCombines elements (for example a series of titles) until a section"
-                + "\n\t\t\treaches a length of n characters."
+                + "\n\t\t\treaches a length of n characters. Only applies to 'by_title' strategy."
                 + "\n\t\tnew_after_n_chars"
-                + "\n\t\t\tCuts off new sections once they reach a length of n characters"
-                + "\n\t\t\ta soft max."
+                + "\n\t\t\tCuts off chunks once they reach a length of n characters; a soft max."
                 + "\n\t\tmax_characters"
                 + "\n\t\t\tChunks elements text and text_as_html (if present) into chunks"
                 + "\n\t\t\tof length n characters, a hard max."
             )
 
         @functools.wraps(func)
         def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> List[Element]:
+            """The decorated function is replaced with this one."""
+
+            def get_call_args_applying_defaults() -> Dict[str, Any]:
+                """Map both explicit and default arguments of decorated func call by param name."""
+                sig = inspect.signature(func)
+                call_args: Dict[str, Any] = dict(**dict(zip(sig.parameters, args)), **kwargs)
+                for param in sig.parameters.values():
+                    if param.name not in call_args and param.default is not param.empty:
+                        call_args[param.name] = param.default
+                return call_args
+
+            # -- call the partitioning function to get the elements --
             elements = func(*args, **kwargs)
-            sig = inspect.signature(func)
-            params: Dict[str, Any] = dict(**dict(zip(sig.parameters, args)), **kwargs)
-            for param in sig.parameters.values():
-                if param.name not in params and param.default is not param.empty:
-                    params[param.name] = param.default
-            if params.get("chunking_strategy") == "by_title":
-                elements = chunk_by_title(
+
+            # -- look for a chunking-strategy argument and run the indicated chunker when present --
+            call_args = get_call_args_applying_defaults()
+
+            if call_args.get("chunking_strategy") == "by_title":
+                return chunk_by_title(
                     elements,
-                    multipage_sections=params.get("multipage_sections", True),
-                    combine_text_under_n_chars=params.get("combine_text_under_n_chars", 500),
-                    new_after_n_chars=params.get("new_after_n_chars", 500),
-                    max_characters=params.get("max_characters", 500),
+                    combine_text_under_n_chars=call_args.get("combine_text_under_n_chars", 500),
```

Comment:
the default parameters also appear in other spots in code -> should we consider making them constants to ensure consistency in all interfaces?

Reply:
I'm going to introduce the constants in a separate PR. It looks like there are several spots outside this immediate area that will need to use them and there may also be ingest-test expection-file changes required too.